ci/prepare-unity.sh

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+
+### This script is used to setup Improbable's internal build machines.
+### If you don't work at Improbable, this may be interesting as a guide to what software versions we use for our
+### automation, but not much more than that.
+
+set -e -u -x -o pipefail
+
+cd "$(dirname "$0")/../"
+
+# Only run inside TeamCity
+if [ -z "${TEAMCITY_CAPTURE_ENV+x}" ]; then
+    exit 0
+fi
+
+source ci/includes/pinned-tools.sh
```

Comment:
try to keep it consistent with the other scripts. We first source all `*.sh` and then add logic. So move the TeamCity check after the `source ...` commands

for the platform-check functions: let's move them to pinned-tools in case we need them somewhere else as well